### ISSUE TITLE
Integrate KEDA scale logic with coming HPA scale to zero features

### ISSUE THREAD

**Issue by jeffhollan** [kedacore/keda#208]
https://github.com/kubernetes/kubernetes/pull/74526

**Comment by mikhailshilkov** [comment]
If I read it correctly, the scale to/from zero feature of HPA has been released in 1.16. Does it mean that it will replace the KEDA way of enabling/disabling the deployment any time soon? Does it make the whole controller obsolete?

**Comment by jeffhollan** [comment]
A few things to note

- While HPA can scale custom metrics to zero, it still needs metrics to scale. So KEDA still relevant in being a single spot for a bunch of metric publishers and authentication with them

- Will also be some time before 1.16 is widely adopted so in interim good to have. 

We should do it but just means one of the things KEDA does today it can offload to HPA, but would still need to do others

**Comment by mikhailshilkov** [comment]
Yes, I'm not questioning KEDA, I guess `the whole controller` was an exaggeration but `handleScaleLoop` can probably go.
Thanks for the quick response!

**Comment by jeffhollan** [comment]
Makes sense. The other thing I want to check out (let me know if others know in meantime) is the polling interval and latency of HPA scaling from 0. In KEDA it’s configurable. Not sure if we’d leave KEDA as an option to still drive if it’s providing some additional features

**Comment by tomkerkhove** [comment]
We should check how this impacts 2.0 /cc @zroubalik

**Comment by zroubalik** [comment]
It's still alpha in kubernetes: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-gates-for-alpha-or-beta-features (search for `HPAScaleToZero`). We could include this later.

**Comment by tomkerkhove** [comment]
Sound like a plan!

**Comment by JorTurFer** [comment]
Does having this issue open make sense?

**Comment by tomkerkhove** [comment]
It depends - Do we align or still have our own approach? If the latter, why?

**Comment by JorTurFer** [comment]
HPA still doesn't support scale to 0, we deactivate the HPA according to official docs, but the scale to zero features aren't available yet in the HPA itself

**Comment by zroubalik** [comment]
This KEP is still active, https://github.com/kubernetes/enhancements/pull/2022

Let's keep this issue opened to track this effort.

**Comment by tomkerkhove** [comment]
Ok; I thought this was shipped already.

**Comment by noahlz** [comment]
Is there anything we (the community) can do to help move this along? Many of us have been waiting on HPA scale-to-zero support for...years.

**Comment by jlemaes** [comment]
https://kubernetes.io/blog/2026/08/26/kubernetes-v1-37-release/#beta-horizontalpodautoscaler-scale-to-zero 1.37 allows scaling to 0 natively. How will this work together with KEDA?

**Comment by JorTurFer** [comment]
so far, KEDA will remain using the operator for scaling to 0 because it allows advanced configuration there. HPA still needs an external metric, so we still need to do almost everything

**Comment by zroubalik** [comment]
> so far, KEDA will remain using the operator for scaling to 0 because it allows advanced configuration there. HPA still needs an external metric, so we still need to do almost everything

💯
